airflow/providers/SUSPENDING_AND_RESUMING_PROVIDERS.rst

```diff
@@ -0,0 +1,150 @@
+ .. Licensed to the Apache Software Foundation (ASF) under one
+    or more contributor license agreements.  See the NOTICE file
+    distributed with this work for additional information
+    regarding copyright ownership.  The ASF licenses this file
+    to you under the Apache License, Version 2.0 (the
+    "License"); you may not use this file except in compliance
+    with the License.  You may obtain a copy of the License at
+
+ ..   http://www.apache.org/licenses/LICENSE-2.0
+
+ .. Unless required by applicable law or agreed to in writing,
+    software distributed under the License is distributed on an
+    "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+    KIND, either express or implied.  See the License for the
+    specific language governing permissions and limitations
+    under the License.
+
+Suspending providers
```

Comment:
@uranusjr : As discussed in https://github.com/apache/airflow/pull/30359#discussion_r1152855962 - here is comprehensive guide on what to do when suspending/resuming.

In most cases it will be just:

a) flip `suspended` to true/false
b) run `breeze ci-image build --upgrade-to-newer-dependencies` - to remove /add dependencies 
c) run pre-commits (several times possibly to update all information and fix any discrepancies reported by pre-commits).

Only in most complex cases (i.e. when we would like to remove `google` for example) we would need to add a bot more conditional imports and `pytest.importskip` - where we are not able to exclude tests/code automatically - but those will be extremely rare and I added examples what to do in this case.